pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -577,11 +577,48 @@ func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) er
 	return nil
 }
 
+// ensureInstanceGroupExists checks whether an instancegroup exists.
+func (r *Reconciler) ensureInstanceGroupExists(instanceGroupName string) error {
+	_, err := r.computeService.InstanceGroupGet(r.projectID, r.providerSpec.Zone, instanceGroupName)
+	if err != nil {
+		return fmt.Errorf("instanceGroupGet request failed: %v", err)
+	}
+	return nil
+}
+
+// ensureInstanceGroupInBackendService checks whether an instancegroup is assigned to a backend service.
+func (r *Reconciler) ensureInstanceGroupInBackendService() error {
+	backendService, err := r.computeService.BackendServiceGet(r.projectID, r.providerSpec.Region, r.backendServiceName())
+	if err != nil {
+		return fmt.Errorf("backendServiceGet request failed: %v", err)
+	}
+
+	for _, backend := range backendService.Backends {
+		if backend.Group == r.FQDNInstanceGroup() {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("couldn't find the instance group %s in a backend service", r.FQDNInstanceGroup())
+}
+
 // registerInstanceToControlPlaneInstanceGroup ensures that the instance is assigned to the control plane instance group of its zone.
 func (r *Reconciler) registerInstanceToControlPlaneInstanceGroup() error {
 	instanceSelfLink := fmtInstanceSelfLink(r.projectID, r.providerSpec.Zone, r.machine.Name)
 	instanceGroupName := r.controlPlaneGroupName()
 
+	if err := r.ensureInstanceGroupExists(instanceGroupName); err != nil {
+		if errRegister := r.registerNewInstanceGroup(); errRegister != nil {
+			return fmt.Errorf("failed to register the new instance group named %s: %v", instanceGroupName, errRegister)
+		}
+	}
+
+	if err := r.ensureInstanceGroupInBackendService(); err != nil {
+		if errPatch := r.updateBackendServiceWithInstanceGroup(); err != nil {
```

Comment:
Same comments as above, but I'd probably expect to move the backend bit into `ensureInstanceGroup` as well so there's just a single new branch from this function